### PR TITLE
Add entity properties to rego evaluation context.

### PR DESCRIPTION
**Pull request by blkt**
# Summary

This change adds the entity's properties as defined by the provider to the REGO evaluation context, making it possible to use them as arguments to e.g. data sources.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [X] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

Manual tests.

# Review Checklist:

- [X] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [X] Checked that related changes are merged.

### CODE REVIEW

File: internal/engine/eval/rego/eval.go [hunk 1]
```diff
@@ -134,14 +138,40 @@ func (e *Evaluator) Eval(
...
 	if err != nil {
 		return nil, fmt.Errorf("error evaluating profile. Might be wrong input: %w", err)
 	}
 
 	return e.reseval.parseResult(rs, entity)
 }
+
+func enrichInputWithEntityProps(
+	input *Input,
+	entity protoreflect.ProtoMessage,
```
Comment (JAORMX) [inline internal/engine/eval/rego/eval.go:161]: What about using an interface to verify if you can get properties? Newer entities would use the EntityInstance protobuf, which is missing, but would also work with that.

File: internal/engine/eval/rego/eval.go [hunk 2]
```diff
@@ -134,14 +138,40 @@ func (e *Evaluator) Eval(
...
+) error {
+	switch entity := entity.(type) {
+	case *minderv1.Repository:
+		input.Properties = entity.Properties
```
Comment (JAORMX) [inline internal/engine/eval/rego/eval.go:165]: Can rego evaluate protobuf? Wouldn't it be easier to use a map[string]any?

Comment (blkt) [reply internal/engine/eval/rego/eval.go:165]: It gets serialized to JSON, so `input` ends up being this (produced using `mindev`)
```
{
  "ingested": null,
  "output_format": "text",
  "profile": {},
  "properties": {
    "github/base_ref": "main",
    "github/clone_url": "https://github.com/blkt-test-org/demo-repo-python.git",
    "github/default_branch": "main",
    "github/deploy_url": "https://api.github.com/repos/blkt-test-org/demo-repo-python/deployments",
    "github/hook_id": 516821355,
    "github/hook_name": "web",
    "github/hook_type": "Repository",
    "github/hook_uiid": "5567daf7-6e06-4494-8a65-993f40ab040d",
    "github/hook_url": "https://api.github.com/repos/blkt-test-org/demo-repo-python/hooks/516821355",
    "github/license": "Apache-2.0",
    "github/primary_language": "",
    "github/pull_author_id": 328978,
    "github/pull_author_login": "blkt",
    "github/pull_number": 1,
    "github/pull_ref": "blkt-patch-1",
    "github/pull_sha": "b751f485143a093cd1808cce6ca83b87c72986c6",
    "github/pull_url": "https://github.com/blkt-test-org/demo-repo-python/pull/1",
    "github/repo_id": 894536760,
    "github/repo_name": "demo-repo-python",
    "github/repo_owner": "blkt-test-org",
    "github/target_url": "https://github.com/blkt-test-org/demo-repo-python.git",
    "is_archived": false,
    "is_fork": true,
    "is_private": false,
    "name": "blkt-test-org/demo-repo-python/1",
    "upstream_id": 2201191245
  }
}
```

File: internal/engine/eval/rego/eval.go (hunk 1 above)
Comment (blkt) [reply internal/engine/eval/rego/eval.go:161]: I used the same type defined [here](https://github.com/mindersec/minder/blob/main/internal/engine/eval/rego/eval.go#L111), but I can map it to that other type/interface if you point me to an example.

File: internal/engine/eval/rego/eval.go (hunk 2 above)
Comment (JAORMX) [reply internal/engine/eval/rego/eval.go:162]: that's awesome and it works for me.

File: internal/engine/eval/rego/eval.go (hunk 1 above)
Comment (JAORMX) [reply internal/engine/eval/rego/eval.go:158]: There is no such interface, you'd need to define a `propertiesFetcher` interface or something of the sort that accepts `GetProperties`

**Review by JAORMX — DISMISSED** [review]

**Comment by blkt** [comment]
sorry @JAORMX, linter was complaining.

**Pull request merged**